### DISCUSSION TITLE
Can I get Tiktok video download url using TikTok-Api?

### DISCUSSION THREAD

**Discussion by Robert0623** [davidteather/TikTok-Api#1156] [Q&A]
Hi everyone,
I want to get the video download url when searching by hashtag using Tiktok-Api(V6.3.0).
I already checked downloadAddr and playAddr, but found that access to this url is denied.

![image](https://github.com/davidteather/TikTok-Api/assets/93371989/17c07e21-188f-4880-8691-4d0d0264299e)

Is there any solution to get video download url?
I would be very thankfull for any kind of help.
Thank you.

**Comment by Arres2** [comment]
![image](https://github.com/davidteather/TikTok-Api/assets/61435563/60857a93-1910-41d7-8b7a-1c9546e8741b)
Seems like the method for doing that is not done yet

**Comment by tilekm** [comment]
I have a one solution, but it has nothing to do with this repo:
`def download_video(url):
    response = requests.get(url, headers=headers)
    spl = response.url.split('/')
    if spl[4] == 'video':
        video_id = spl[5].split('?')[0]
        request_url = f'https://www.tikwm.com/video/media/play/${video_id}.mp4'
        response = requests.get(request_url, headers=headers)
        video_link = response.url
        return video_link
    else:
        return False`

**Reply by tudragon154203** [reply]
You are a hero! Still works 10.2025

**Comment by joaovvbp** [comment]
Check out the [latest release](https://github.com/davidteather/TikTok-Api/releases/tag/V6.4.0)

> @bendavidsteel
[Restored functionality for download bytes method (](https://github.com/davidteather/TikTok-Api/pull/1178/commits/9bc381e63b2f3e5692e039599f4717c8b50b6266)[https://github.com/davidteather/TikTok-Api/pull/1174[)](https://github.com/davidteather/TikTok-Api/pull/1178/commits/9bc381e63b2f3e5692e039599f4717c8b50b6266](https://github.com/davidteather/TikTok-Api/pull/1174%5B)%5D(https://github.com/davidteather/TikTok-Api/pull/1178/commits/9bc381e63b2f3e5692e039599f4717c8b50b6266))

**Reply by joaovvbp** [reply]
I also found this to work previously:

```
video_data_raw = video.as_dict
download_url = video_data_raw['video']['bitrateInfo'][0]['PlayAddr']['UrlList'][-1]
```

**Comment by niletechnology1** [comment]
[Download Video TikTok](https://www.downloadvideotiktok.online/) tool is a powerful and user-friendly solution for downloading TikTok content quickly and without watermark. Whether you want to Download Video TikTok Tanpa Watermark, save clips in HD, or convert videos into audio, this tool supports everything. You can easily Download Video TikTok MP4 for high-quality playback or Download Video TikTok MP3 for music extraction. It also allows users to Download TikTok Stories and trending clips with just a few clicks.
This TikTok Downloader works online, so there’s no need to install apps or create an account. Simply paste the video link and start your TikTok Video Download instantly. The process is fast, free, and compatible with all devices. Whether you’re looking to Download TikTok Video or save content for offline use, this TikTok Download tool provides a smooth and reliable experience for everyone.
